### PR TITLE
fix: make lookup_path and lookup_label comply with Interface Spec

### PR DESCRIPTION
**Pull request by mraszyk**

### CODE REVIEW

File: src/hash_tree.rs [hunk 1]
```diff
@@ -351,21 +354,21 @@ impl<'a> HashTreeNode<'a> {
...
                 match left_label {
                     // On continue or unknown, look on the right side of the fork.
                     // If it cannot be found on the right, return Unknown though.
-                    LookupLabelResult::Continue | LookupLabelResult::Unknown => {
+                    LookupLabelResult::Greater | LookupLabelResult::Unknown => {
                         let right_label = nodes.1.lookup_label(label);
                         match right_label {
-                            LookupLabelResult::Absent => {
+                            LookupLabelResult::Less => {
                                 if matches!(left_label, LookupLabelResult::Unknown) {
```
Comment (roman-kashitsyn) [inline src/hash_tree.rs:372]: This code looks somewhat weird: we first match both `Greater` and `Unknown`, and then we're trying to figure out which one it was. I'd just split the "or" clause; it looks cleaner, IMO.

**Review by roman-kashitsyn — COMMENTED** [review]
Could we add a couple of test cases?

**Comment by mraszyk** [comment]
I've added some more tests. Several of them fail before this PR.

**Review by roman-kashitsyn — APPROVED** [review]
LGTM, thank you very much for fixing this!

**Pull request merged**